### PR TITLE
Fixed resourceRead field from settings to read dirty data when contains the union-all opeartor in hive-sql.

### PR DESCRIPTION
**Pull request by AbelChan1989**
HI,

   The pull request tries to solve the #issue 1084 (https://github.com/elastic/elasticsearch-hadoop/issues/1084).

   I find the resouceRead field from settings can be contaminated when a hive-sql contains union-all opeartor. So it may lead to a wrong output result. 

   For fixed this problem, this resourceRead value should be obtained from the Hive TBLPROPERTIES.

Thanks for a great product.

### CODE REVIEW

**Comment by karmi** [comment]
Hi @AbelChan1989, we have found your signature in our records, but it seems like you have signed with a different e-mail than the one used in yout Git [commit](https://github.com/elastic/elasticsearch-hadoop/pull/1085.patch). Can you please add both of these e-mails into your Github profile (they can be hidden), so we can match your e-mails to your Github profile?

**Comment by AbelChan1989** [comment]
@karmi OK, done.  thx

**Comment by jbaiera** [comment]
Hi @AbelChan1989, thanks for opening a PR. The CI job here has failed due to an unrelated issue on the master branch. I'll kick it off again once it's resolved.

**Comment by jbaiera** [comment]
I was speaking more along the lines of adding a test case to [the integration test](https://github.com/elastic/elasticsearch-hadoop/blob/master/hive/src/itest/java/org/elasticsearch/hadoop/integration/hive/AbstractHiveSearchTest.java). I just want to make sure that going forward, new work on Hive doesn't regress this bug fix. Thanks again for working on the PR!

**Comment by AbelChan1989** [comment]
HI @jbaiera ,

   OK, thanks for your reminding. I have added the test cases: org.elasticsearch.hadoop.integration.hive.AbstractHiveSearchTest#basicUnion.

File: hive/src/main/java/org/elasticsearch/hadoop/hive/EsHiveInputFormat.java [hunk 1]
```diff
@@ -106,8 +106,14 @@ public Path getPath() {
             // Only set the fields if we aren't asking for raw JSON
             settings.setProperty(InternalConfigurationOptions.INTERNAL_ES_TARGET_FIELDS, StringUtils.concatenate(HiveUtils.columnToAlias(settings), ","));
         }
-        // set read resource
-        settings.setResourceRead(settings.getResourceRead());
+
+        // set read resource, with es.resource in Hive TBLPROPERTIES
+        // avoid resourceRead to read dirty data when exist the union-all opeartor in sql.  
+//	String esResource = job.get(ConfigurationOptions.ES_RESOURCE);
```
Comment (jbaiera) [inline hive/src/main/java/org/elasticsearch/hadoop/hive/EsHiveInputFormat.java]: Please remove this commented out code

**Comment by AbelChan1989** [comment]
HI @jbaiera ,
I am so sorry for the late reply. I have removed the commented code and submitted a new PR, please review, thanks.

**Comment by jbaiera** [comment]
Thanks for removing the comment. Sorry for the slow response here. This LGTM. Thanks for the PR!

**Pull request merged**